### ISSUE TITLE
Add support for WAV streaming

### ISSUE THREAD

**Issue by toby-yoyo** [YoYoGames/GameMaker-Bugs#8429]
### Is your feature request related to a problem?

At present, users must use Ogg/Vorbis for streaming audio. This restriction presents a few issues:
- It incurs significant CPU overhead at runtime.
- It the limits audio quality of streamed audio. 
- It complicates workflows for projects, especially for those where compression is not desired.
- It limits flexibility in supporting externally-sourced audio (e.g. mods).

Adding support for WAV streaming will provide several benefits:
- Users will gain a way of playing high-quality tracks without preloading them into memory.
- Users will gain a more CPU-efficient way to stream audio (when compared to compressed formats).
- Users will gain a straightforward method of creating sounds from WAV files at runtime, without needing third-party libraries.
- Users will gain greater flexibility in playing externally sourced audio files.
- The IDE will regain the ability to stream uncompressed files, which was lost when the runtime audio engine was integrated.

### Describe the solution you'd like

**IDE**
- Add an `Uncompressed - Streamed` option to the sound editor/inspector’s attributes section.
- Ensure that sound assets in the IDE are streamed when played.

**Runtime**
- Allow `audio_create_stream` to accept WAV files for uncompressed streaming. The formats of WAV files accepted should be as wide as possible.

### Describe alternatives you've considered

One could use GML to write systems that preload or stream WAV files at runtime, but this would be complex to implement well. Adding support for native WAV streaming therefore seems like a much more user-friendly solution.

### Additional context

#2862 overlaps with the runtime side of this issue, but it may be made redundant in the future, depending on the exact nature of GMRT's intended dynamic asset creation capabilities. It is harder to say the same for this particular request, which also adds functionality to the IDE.

**Comment by rwkay** [comment]
WAV streaming will add significantly to disc bandwidth required, it may also fall foul of requirements on Switch for the number of reads required to maintain the bandwidth. WAV is significantly bigger and requires more data for the same length of music and to fill the buffer (before another read is issued).

We need to check the restrictions on Switch before we add an option like this.

**Comment by DragoniteSpam** [comment]
Is this something you need to do often? I can't say I've ever seen a game (AAA or otherwise) streaming wav files from the disk, and I can't imagine the tradeoff of disk space would be worth it for an audio file that's more than a few seconds long for most people.

**Comment by toby-yoyo** [comment]
No, I imagine this wouldn't be used too often at runtime. I had intended to have this in for the IDE primarily, in order not to avoid loading assets into memory in order to play them (which is what happens now), and also to have the IDE more accurately reflect what each asset will sound like at runtime based on its settings in the sound editor. Since the audio library is now shared between the IDE and runtime I thought we might as well turn it into a user-accessible feature if it were added. But if there is no desire to have an option like this, then we can focus our efforts on something more useful.

Alternatively, we could refocus this FR around a different lossless codec (e.g. FLAC) to reduce the size of converted streams and the disk bandwidth used.

**Comment by toby-yoyo** [comment]
Done as part of #8632.

**Comment by sihammill** [comment]
Verified #8632 in Beta IDE v2024.1400.0.796  Runtime v2024.1400.0.803 and so this is also verified as part of that.
